### PR TITLE
DRY notification formatting

### PR DESCRIPTION
**Pull request by EmilioCristalli**
I think there was duplicated code to format the notification in some notifiers.
This PR tries to move some of that logic to its own class to make it easier to maintain and test

### CODE REVIEW

File: lib/exception_notifier/modules/formatter.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      end
+    end
+
+    #
+    # A *NoMethodError* occurred.
+    # 3 *NoMethodError* occurred.
+    # A *NoMethodError* occurred in *home#index*.
+    #
+    def subtitle
+      errors_text = errors_count > 1 ? errors_count : 'A'
```
Comment (FLarra) [inline lib/exception_notifier/modules/formatter.rb]: What about having errors_text differentiate `A/An`?

I'd do:

```rb
errors_text = if errors_count > 1
                errors_count
              else
                exception_class.to_s =~ /^[aeiou]/i ? 'An' : 'A'
              end
```

What do you think?

File: test/exception_notifier/mattermost_notifier_test.rb [hunk 2]
```diff
@@ -1,102 +1,184 @@
...
+        '@channel',
+        '### ⚠️ Error occurred in test ⚠️',
+        'A *ArgumentError* occurred.',
+        '*foo*'
+      ].join("\n"),
+      username: 'Exception Notifier'
+    }
+  end
+
+  def defaul_headers
```
Comment (FLarra) [inline test/exception_notifier/mattermost_notifier_test.rb]: Minor typo:

```suggestion
  def default_headers
```

File: test/exception_notifier/google_chat_notifier_test.rb [hunk 3]
```diff
@@ -160,12 +160,12 @@ def header
     [
       '',
       'Application: *dummy*',
-      'An *ArgumentError* occured.',
+      'A *ArgumentError* occurred.',
```
Comment (FLarra) [inline test/exception_notifier/google_chat_notifier_test.rb]: I'd adjust the code to return `An` in this case.

File: lib/exception_notifier/modules/formatter.rb (hunk 1 above)
Comment (EmilioCristalli) [reply lib/exception_notifier/modules/formatter.rb]: good idea, changed here df864ea68405a49a213fa0d0ca839983af31d0cc
thanks!

File: test/exception_notifier/google_chat_notifier_test.rb (hunk 3 above)
Comment (EmilioCristalli) [reply test/exception_notifier/google_chat_notifier_test.rb]: changed here: df864ea68405a49a213fa0d0ca839983af31d0cc

File: test/exception_notifier/mattermost_notifier_test.rb (hunk 2 above)
Comment (EmilioCristalli) [reply test/exception_notifier/mattermost_notifier_test.rb]: thanks!, fixed here 290d3f9d0a45741d4817ac2c68c55d62a07f8408

**Comment by EmilioCristalli** [comment]
@FLarra thanks for the review!
I think I addressed all your comments, let me know if i missed anything!

**Comment by FLarra** [comment]
Yes Sir! 🎉 Amazing! 👏 👏 This is great! 

Thank you

**Pull request merged**